.github/workflows/make.yml

```diff
@@ -0,0 +1,117 @@
+name: Build Emulation Cores
+
+on:
+  pull_request:
+    branches: [ "master" ]
+  push:
+    branches: [ "master" ]
+
+env:
+  CC: clang-18
+  CXX: clang++-18
+
+jobs:
+
+  build-waterbox:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v4
+    - name: Installing clang 18
+      run: wget https://apt.llvm.org/llvm.sh;
+          chmod u+x llvm.sh;
+          sudo ./llvm.sh 18;
+          clang-18 --version;    
+    - name: Getting submodule(s)
+      working-directory: ./waterbox/
+      run: git submodule update --init --recursive musl
+    - name: Building musl
+      working-directory: ./waterbox/musl
+      run: ./wbox_configure.sh;
+           ./wbox_build.sh;
+    - name: Building emulibc
+      working-directory: ./waterbox/emulibc
+      run: make
+    - name: Building libco
+      working-directory: ./waterbox/libco
+      run: make
+    - name: Building libcxx
+      working-directory: ./waterbox/libcxx
+      run: ./do-everything.sh
+    - name: Building nyma
+      working-directory: ./waterbox/nyma
+      run: ./build-and-install-zlib.sh
+    - name: Uploading compiled waterbox
+      uses: actions/upload-artifact@v4
+      with:
+        name: compiled-waterbox
+        path: |
+          waterbox/libco
+          waterbox/musl
+          waterbox/emulibc
+          waterbox/libcxx
```

Comment:
The `waterbox/libcxx` path is not required to be packed into the artifact; this directory only contains intermediate build files.